### ISSUE TITLE
brew doctor doesn't work with `HOMEBREW_INSTALL_FROM_API` and untap homebrew/core

### ISSUE THREAD

**Issue by fanvinga** [Homebrew/brew#12356]
### `brew config` output

```shell
HOMEBREW_VERSION: 3.3.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9daf5e2d28396c7a4379605e89e029d9fccb41e6
Last commit: 2 hours ago
Core tap: N/A
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.MstZqkjrdb/org.xquartz:0
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.33.1 => /opt/homebrew/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-arm64
CLT: 13.1.0.0.1.1633545042
Xcode: N/A
XQuartz: 2.8.1 => /opt/X11
Rosetta 2: false
```

### `brew doctor` output

```shell
==> Checking check_access_directories
==> Checking check_brew_git_origin
/opt/homebrew/Library/Homebrew/shims/shared/git --version
==> Checking check_broken_sdks
==> Checking check_cask_environment_variables
==> Checking check_cask_install_location
==> Checking check_cask_load_path
==> Checking check_cask_quarantine_support
==> Checking quarantine support
/usr/bin/xattr -h
/usr/bin/swift /opt/homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Checking check_cask_software_versions
==> Checking check_cask_staging_location
==> Checking check_cask_taps
==> Checking check_cask_xattr
/usr/bin/xattr -h
==> Checking check_casktap_integrity
==> Checking check_clt_minimum_version
==> Checking check_clt_up_to_date
==> Checking check_coretap_integrity
==> Checking check_deleted_formula
Error: undefined method `core_tap?' for nil:NilClass
/opt/homebrew/Library/Homebrew/diagnostic.rb:887:in `block in check_deleted_formula'
/opt/homebrew/Library/Homebrew/diagnostic.rb:886:in `map'
/opt/homebrew/Library/Homebrew/diagnostic.rb:886:in `check_deleted_formula'
/opt/homebrew/Library/Homebrew/cmd/doctor.rb:64:in `block in doctor'
/opt/homebrew/Library/Homebrew/cmd/doctor.rb:57:in `each'
/opt/homebrew/Library/Homebrew/cmd/doctor.rb:57:in `doctor'
/opt/homebrew/Library/Homebrew/brew.rb:110:in `<main>'
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

ensure everything works under `HOMEBREW_INSTALL_FROM_API`

### What happened (include all command output)?

I put `export HOMEBREW_INSTALL_FROM_API=true` in my `~/.zshrc`.Then I run `brew update` for updating homebrew to 3.3.2.After updating I found that `brew doctor` doesn't work.

### What did you expect to happen?

`brew doctor` returns 
`Your system is ready to brew.`

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1.Update homebrew to 3.3.2 by `brew update`
2.Run `brew doctor`
```

**Comment by fanvinga** [comment]
Here's my `brew tap` output
```
╭─ ~/Desktop [ 14:55:48 ] ✗
╰─ brew tap --debug
teamookla/speedtest
```

**Comment by carlocab** [comment]
CC @Rylan12

**Comment by Rylan12** [comment]
Thanks for the report! Not totally sure what happened here... Thought I fixed this in #12333 but, clearly, I didn't thoroughly test that part of the PR 😭😅

**Comment by fanvinga** [comment]
> 谢谢你的报告！不完全确定这里发生了什么......我以为我在#12333中修复了这个，但显然，我没有彻底测试公关的那部分😭

Thank you very much for your quick response, please don't feel frustrated about the omission, I think you have done a great job! I will try to `brew doctor` again after the pr merged.
